convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -421,487 +358,343 @@ def test_call_yum_cmd_with_repo_overrides(self):
 
-        # Second, test utility exiting when unable to replace the kernel
-        utils.download_pkg.to_return = "/path/to.rpm"
-        utils.run_subprocess.ret_code = 1
-        version = "4.7.4-200.fc24"
-        self.assertRaises(SystemExit, pkghandler.replace_non_rhel_installed_kernel, version)
 
-    def test_get_kernel(self):
-        kernel_version = list(pkghandler.get_kernel(YUM_KERNEL_LIST_OLDER_NOT_AVAILABLE))
+class TestIsRHELKernelInstalled:
+    def test_is_rhel_kernel_installed_no(self, monkeypatch):
+        monkeypatch.setattr(pkghandler, "get_installed_pkgs_by_fingerprint", lambda x, name: [])
 
-        self.assertEqual(kernel_version, ["4.7.4-200.fc24", "4.7.4-200.fc24"])
+        assert not pkghandler.is_rhel_kernel_installed()
 
-    @unit_tests.mock(pkghandler, "is_rhel_kernel_installed", lambda: True)
-    def test_verify_rhel_kernel_installed(self):
-        pkghandler.verify_rhel_kernel_installed()
+    def test_is_rhel_kernel_installed_yes(self, monkeypatch):
+        monkeypatch.setattr(
+            pkghandler,
+            "get_installed_pkgs_by_fingerprint",
+            lambda x, name: ["kernel"],
+        )
 
-    @unit_tests.mock(pkghandler, "is_rhel_kernel_installed", lambda: False)
-    def test_verify_rhel_kernel_installed_not_installed(self):
-        self.assertRaises(SystemExit, pkghandler.verify_rhel_kernel_installed)
+        assert pkghandler.is_rhel_kernel_installed()
 
-    @unit_tests.mock(pkghandler, "get_installed_pkgs_by_fingerprint", lambda x, name: [])
-    def test_is_rhel_kernel_installed_no(self):
-        self.assertFalse(pkghandler.is_rhel_kernel_installed())
 
-    @unit_tests.mock(
-        pkghandler,
-        "get_installed_pkgs_by_fingerprint",
-        lambda x, name: ["kernel"],
+class TestFixInvalidGrub2Entries:
+    @pytest.mark.parametrize(
+        ("version", "arch"),
+        (
+            (Version(7, 0), "x86_64"),
+            (Version(8, 0), "s390x"),
+        ),
     )
-    def test_is_rhel_kernel_installed_yes(self):
-        self.assertTrue(pkghandler.is_rhel_kernel_installed())
+    def test_fix_invalid_grub2_entries_not_applicable(self, version, arch, caplog, monkeypatch):
+        monkeypatch.setattr(system_info, "version", version)
+        monkeypatch.setattr(system_info, "arch", arch)
 
-    @unit_tests.mock(system_info, "version", Version(7, 0))
-    @unit_tests.mock(system_info, "arch", "x86_64")
-    @unit_tests.mock(pkghandler.logging, "getLogger", GetLoggerMocked())
-    def test_fix_invalid_grub2_entries_not_applicable(self):
         pkghandler.fix_invalid_grub2_entries()
-        self.assertFalse(len(pkghandler.logging.getLogger.info_msgs), 1)
 
-        system_info.version = Version(8, 0)
-        system_info.arch = "s390x"
-        pkghandler.fix_invalid_grub2_entries()
-        self.assertFalse(len(pkghandler.logging.getLogger.info_msgs), 1)
+        assert not [r for r in caplog.records if r.levelname != "DEBUG"]
```

Comment:
Not sure I follow this fully but it is also 3 am so

Reply:
The new version is slightly stricter.  Although, neither one is very good.  The test is checking that if the platform never needs to try to fix up the grub records, then the `fix_invalid_grub2_entries()` function will exit before it has a chance to do anything.  In both cases, it does that by looking to see if log messages that the function would normally output were emitted.  The old test only looked for an info message (the first statement after the early exit is: `loggerinst.info("Fixing GRUB boot loader entries.")`.)  The new test looks for anything above debug.

If I was changing convert2rhel runtime code in this PR, I would probably have changed `fix_invalid_grub2_entries()` to emit a message when it exits early. Then look for the presence of that rather than looking for the absence of any message.